lib/spack/docs/module_file_tutorial.rst

```diff
@@ -0,0 +1,960 @@
+  prepend_path("LD_LIBRARY_PATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-4.8/gcc-6.2.0-twd5nqg33hrrssqclcfi5k42eccwxz5u/lib64")
+  setenv("GCC_ROOT","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-4.8/gcc-6.2.0-twd5nqg33hrrssqclcfi5k42eccwxz5u")
+  setenv("CC","gcc")
+  setenv("CXX","g++")
+  setenv("F90","gfortran")
+  setenv("FC","gfortran")
+  setenv("F77","gfortran")
+  conflict("gcc")
+  help([[The GNU Compiler Collection includes front ends for C, C++, Objective-C,
+  Fortran, and Java.
+  ]])
+
+  $ module show openmpi
+  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+     /home/mculpo/wdir/spack/share/spack/modules/linux-Ubuntu14-x86_64/openmpi/2.0.1-gcc-6.2.0:
+  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  whatis("openmpi @2.0.1 ")
+  prepend_path("PATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w/bin")
+  prepend_path("CMAKE_PREFIX_PATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w/")
+  prepend_path("LD_LIBRARY_PATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w/lib")
+  prepend_path("PKG_CONFIG_PATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w/lib/pkgconfig")
+  prepend_path("MANPATH","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w/share/man")
+  setenv("SLURM_MPI_TYPE","pmi2")
+  setenv("OMPI_MCA_BTL_OPENIB_WARN_DEFAULT_GID_PREFIX","0")
+  setenv("OPENMPI_ROOT","/home/mculpo/wdir/spack/opt/spack/linux-Ubuntu14-x86_64/gcc-6.2.0/openmpi-2.0.1-s3qbtbyh3y5y4gkchmhcuak7th44l53w")
+  conflict("openmpi")
+  help([[The Open MPI Project is an open source Message Passing Interface
+  implementation that is developed and maintained by a consortium of
+  academic, research, and industry partners. Open MPI is therefore able to
+  combine the expertise, technologies, and resources from all across the
+  High Performance Computing community in order to build the best MPI
+  library available. Open MPI offers advantages for system and software
+  vendors, application developers and computer science researchers.
+  ]])
+
+
+---------------------
+Autoload dependencies
+---------------------
+
+Spack can also generate module files that contain code to autoload
+the dependencies automatically. You can for instance generate python
+modules that autoload their dependencies adding the ``autoload`` directive:
+
```

Comment:
Neat!  Does this work with Environment-Modules or only Lmod?  If both, please specify: because auto-loading dependencies is not an advertised feature of Environment-Modules.